Add userscript for WCPay Live Branch

htdat opened the pull request:
Fixes https://github.com/Automattic/woocommerce-payments-meta/issues/92

TODO for myself: 

- [x] Finish the user script.
- [x] Update the comment body in [pr-build-live-branch.yml](https://github.com/Automattic/woocommerce-payments/blob/033c3a5b6bfe0e11f4c1522073d8cc943f1049f5/.github/workflows/pr-build-live-branch.yml#L116-L125)
- [x] Write a P2 post internally. 

#### Changes proposed in this Pull Request

- Add a user script like [Jetpack](https://github.com/Automattic/jetpack/tree/trunk/tools/jetpack-live-branches) and [WooCommerce core](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-beta-tester/userscripts). 
- Update the comment in the pr-build-live-branch.yml workflow to give more ways and instructions.

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

#### Testing the user script 

- Check this README.md file https://github.com/Automattic/woocommerce-payments/tree/add/jn-userscript-for-live-branch/bin/wcpay-live-branches - when this PR is merged, it will be live at https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches
- Follow up the README.md file but the correct link for now would be https://github.com/Automattic/woocommerce-payments/tree/add/jn-userscript-for-live-branch/bin/wcpay-live-branches
- Refresh this pull request page, and see something like the following in this PR description, select some options and spin up a site :rocket: 

![Markup on 2023-03-21 at 11:40:01](https://user-images.githubusercontent.com/10045087/226525271-e75bb729-4fc1-4792-9989-ef736283fc01.png)

#### Check the comment and Jetpack Beta 

- Check the PR comment https://github.com/Automattic/woocommerce-payments/pull/5801#issuecomment-1473294235
- May test the links there and try Jetpack Beta. 

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] NO NEED - Covered with tests (or have a good reason not to test in description ☝️)
- [x] NO NEED - Tested on mobile (or does not apply)

**Post merge**

NO NEED. 

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Ap
plicable'
-->

htdat committed:
Add workflow "Build live branch for PR"

htdat committed:
Update PLUGIN_DATA for step `bump-dev-version`

htdat committed:
Update bump-dev-version in step "Inform Beta Download webhook"

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Fix $DEV_VERSION

htdat committed:
Update the structure of artifacts for Jetpack Beta Builder

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Add step - Add/update the information as a comment in the PR

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Remove the specific condition for pull_request

htdat committed:
add changelog

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Update the correct commit SHA in PR.

Subtle difference: https://github.com/orgs/community/discussions/26325

htdat committed:
Extract get-dev-version and get-plugin-data

htdat committed:
Update the comment.

htdat committed:
Add first files

htdat committed:
Update the workflow name and changelog entry

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Merge branch 'add/5624-live-branch-support' into add/jn-userscript-for-live-branch

htdat committed:
Update code specifically for wcpay

htdat committed:
Remove the link when the PR is merged.

htdat committed:
Add changelog

htdat committed:
Update the PR comment in the workflow to introduce this feature.

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5801` or branch name `add/jn-userscript-for-live-branch` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/jn-userscript-for-live-branch) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: a4b601d7143b3d3aae545b099bcf12460fc4e63a
- Build time: 2023-03-23 04:33:04 UTC

_Note: the build is updated when a new commit is pushed to this PR._

htdat committed:
Add text for the JN link.

htdat committed:
Merge branch 'develop' into add/5624-live-branch-support

htdat committed:
Merge branch 'add/5624-live-branch-support' into add/jn-userscript-for-live-branch

htdat committed:
Merge branch 'develop' into add/jn-userscript-for-live-branch

# Conflicts:
#	.github/workflows/pr-build-live-branch.yml

htdat committed:
Update the workflow comment.

htdat committed:
Fix the yaml syntax.

htdat committed:
Merge branch 'develop' into add/jn-userscript-for-live-branch

htdat committed:
Update comment in the workflow

htdat commented:
Ready for the review!

htdat committed:
Merge branch 'develop' into add/jn-userscript-for-live-branch

htdat committed:
Use 7 chars for commit hash in DEV_SUFFIX_VERSION

htdat committed:
Comment text: re-arrange options' positions, change some texts

htdat commented on .github/workflows/pr-build-live-branch.yml:
```diff
@@ -113,12 +113,23 @@ jobs:
+         
           :rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=${PR_HEAD_REF}) :rocket:
+          
+          :information_source: [Install this Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.
```
Until this PR is merged, use this link

https://github.com/Automattic/woocommerce-payments/tree/add/jn-userscript-for-live-branch/bin/wcpay-live-branches

jbordonado replied:
```suggestion
          :information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.
```

htdat commented on .github/workflows/pr-build-live-branch.yml:
```diff
@@ -38,7 +38,7 @@ jobs:
           PR_HEAD_SHA: ${{ github.event.pull_request.head.sha }}
         run: |
-          DEV_SUFFIX_VERSION="dev_PR-${PR_NUMBER}_commit-${PR_HEAD_SHA}"
+          DEV_SUFFIX_VERSION="dev_PR-${PR_NUMBER}_commit-${PR_HEAD_SHA:0:7}"
```
A small change to display [the shorten commit hash](https://user-images.githubusercontent.com/10045087/226828532-c0dc372a-39b3-43e5-924d-f42977f520e1.png) rather than [the full commit hash string](https://user-images.githubusercontent.com/10045087/226828410-b87b574f-5871-4575-a87a-a3f2982476b0.png) in Jetpack Beta UI.

htdat commented on bin/wcpay-live-branches/README.md:
```diff
@@ -0,0 +1,18 @@
+### Installation
+
+* Install Tampermonkey or equivalent user script extension.
+* Click https://github.com/Automattic/woocommerce-payments/raw/develop/bin/wcpay-live-branches/wcpay-live-branches.user.js
```
Use this link until this PR is merged. 
https://github.com/Automattic/woocommerce-payments/raw/add/jn-userscript-for-live-branch/bin/wcpay-live-branches/wcpay-live-branches.user.js

jbordonado reviewed (approved):
Great work!

The only thing is that I initially missed the new section in the PR description, so I was wondering where it was or if I had not installed the script correctly.
Maybe we can make it more prominent? A horizontal line above (and not below) and some colors?

Minor thing, it seems good to go otherwise (I haven't looked at all the jQuery code, to be honest, I'll have to trust that it works fine for all cases)

htdat committed:
Update .github/workflows/pr-build-live-branch.yml

Co-authored-by: Jordan Bordonado <28830738+jbordonado@users.noreply.github.com>

htdat committed:
Merge branch 'develop' into add/jn-userscript-for-live-branch

htdat committed:
Make the new section more  prominent in the PR description

htdat commented:
@jbordonado - Great feedback! 

> The only thing is that I initially missed the new section in the PR description, so I was wondering where it was or if I had not installed the script correctly.
Maybe we can make it more prominent? A horizontal line above (and not below) and some colors?

Commit a4b601d7143b3d3aae545b099bcf12460fc4e63a
I've addressed it like this, also added a new bullet in README.md to clarify what users should expect. 

![prominent-new-section](https://user-images.githubusercontent.com/10045087/227102466-6c9e5918-07ae-42cd-9dd9-20d1e355091d.png)

htdat commented:
This PR has nothing to do with two failure checks below so I am going ahead to merge it: 

- Compatibility - WC, WP and PHP / Environment - WC beta (beta, latest, latest, 8.0) (pull_request) Failing after 32s
- Compatibility - WC, WP and PHP / Environment - WC beta (beta, latest, latest, 8.1) (pull_request) Failing after 35s
